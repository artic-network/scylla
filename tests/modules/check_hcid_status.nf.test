nextflow_workflow {

    name "HCID status module tests"
    script "modules/check_hcid_status.nf"
    workflow "check_hcid_status"
    tag "unit"
    tag "ci"

    test("cchf_detection_success") {

        when {
            params {
                
            }
            workflow {
                """
                input[0] = Channel.of(
                    tuple(
                        "cchf_test",
                        "PlusPF",
                        file("${projectDir}/tests/test-data/hcid/3052518/cchf_kraken_report.txt", checkIfExists: true)
                    )
                )
                input[1] = Channel.of(
                    tuple(
                        "cchf_test",
                        file("${projectDir}/tests/test-data/hcid/3052518/cchf_mock.fastq", checkIfExists: true)
                    )
                )
                input[2] = file("${projectDir}/tests/test-data/hcid/3052518/tax_dump", checkIfExists: true)
                """
            }
        }

        then {
            def warning_json = path("${outputDir}/cchf_test/qc/3052518.warning.json").json
            assertAll(
                { assert workflow.success },
                { assert path("${outputDir}/cchf_test/qc/hcid.counts.csv").exists() },
                { assert path("${outputDir}/cchf_test/qc/3052518.warning.json").exists() },
                { assert snapshot(workflow.out, path("${outputDir}/cchf_test/qc/hcid.counts.csv")).match() },
                
                // check core content in addition to snapshot
                { assert warning_json.classified_count == 10 },
                { assert warning_json.mapped_count == 10 },
                // hcid references 
                { assert warning_json.mapped_details.contains("NC_005300.2") },
                { assert warning_json.mapped_details.contains("NC_005301.3") },
                { assert warning_json.mapped_details.contains("NC_005302.1") },
            )
            
        }

    }

    test("cchf_detection_fail") {

        when {
            params {

            }
            workflow {
                """
                input[0] = Channel.of(
                    tuple(
                        "cchf_test",
                        "PlusPF",
                        file("${projectDir}/tests/test-data/hcid/3052518/cchf_kraken_report_blank.txt", checkIfExists: true)
                    )
                )
                input[1] = Channel.of(
                    tuple(
                        "cchf_test",
                        file("${projectDir}/tests/test-data/hcid/3052518/blank.fastq", checkIfExists: true)
                    )
                )
                input[2] = file("${projectDir}/tests/test-data/hcid/3052518/tax_dump", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert path("${outputDir}/cchf_test/qc/hcid.counts.csv").exists() },
                { assert !path("${outputDir}/cchf_test/qc/3052518.warning.json").exists() },
                { assert snapshot(workflow.out, path("${outputDir}/cchf_test/qc/hcid.counts.csv")).match() }
            )
        }

    }

}
