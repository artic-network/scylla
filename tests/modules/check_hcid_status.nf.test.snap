{
    "cchf_detection_fail": {
        "content": [
            {
                "0": [
                    
                ],
                "warning_ch": [
                    
                ]
            },
            "hcid.counts.csv:md5,80d4f9e64b8160d3ef44f476a7833a91"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-15T13:14:27.963809838"
    },
    "cchf_detection_success": {
        "content": [
            {
                "0": [
                    [
                        "cchf_test",
                        "3052518.warning.json:md5,1b59b4c73ec5eb7a87a2e6b1cc810e9a"
                    ]
                ],
                "warning_ch": [
                    [
                        "cchf_test",
                        "3052518.warning.json:md5,1b59b4c73ec5eb7a87a2e6b1cc810e9a"
                    ]
                ]
            },
            "hcid.counts.csv:md5,c45ab01001988dc88e4469ae29a92448"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-15T13:13:51.203986115"
    }
}